models/DataObject/ClassDefinition/Helper/Dao.php

```diff
@@ -172,11 +176,28 @@ protected function removeIndices(string $table, array $columnsToRemove, array $p
         if ($columnsToRemove) {
             $lowerCaseColumns = array_map('strtolower', $protectedColumns);
             foreach ($columnsToRemove as $value) {
-                if (!in_array(strtolower($value), $lowerCaseColumns)) {
-                    Helper::queryIgnoreError($this->db, 'ALTER TABLE `'.$table.'` DROP INDEX `u_index_'. $value . '`;');
+                if (!in_array(strtolower($value), $lowerCaseColumns) && $this->indexExists($table, 'p_index_', $value)) {
+                    $this->db->executeQuery('ALTER TABLE `'.$table.'` DROP INDEX `u_index_'. $value . '`;');
                 }
             }
             $this->resetValidTableColumnsCache($table);
         }
     }
+
+    /**
+     * For MariaDB, it would be possible to use 'ADD/DROP INDEX IF EXISTS' but this is not supported by MySQL
+     */
+    protected function indexExists(string $table, string $prefix, mixed $columnName): bool
+    {
+        $exist = $this->db->fetchFirstColumn(
+            "SELECT COUNT(*) FROM information_schema.statistics WHERE table_name = '${table}' and index_name = '${prefix}${columnName}' and table_schema = database();"
```

Comment:
I think we should use consistent casing for keywords.

```suggestion
            "SELECT COUNT(*) FROM information_schema.statistics WHERE table_name = '${table}' AND index_name = '${prefix}${columnName}' AND table_schema = DATABASE();"
```